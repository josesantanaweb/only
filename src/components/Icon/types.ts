export interface IconProps {
  /** Sets name icon */
  name:
    | "home"
    | "add"
    | "arrow-right"
    | "arrow-left"
    | "arrow-up"
    | "arrow-down"
    | "bell"
    | "cash"
    | "bookmark"
    | "chat"
    | "check"
    | "copy"
    | "dots"
    | "filter"
    | "lock"
    | "lists"
    | "messages"
    | "more"
    | "notifications"
    | "profile"
    | "search"
    | "send"
    | "star"
    | "suscriptions"
    | "tag";
  /** Sets color icon */
  colorIcon?: string;
  /** Sets size icon */
  size?: string;
}
