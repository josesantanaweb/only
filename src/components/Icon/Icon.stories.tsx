import React from "react";

import { ComponentStory, ComponentMeta } from "@storybook/react";

import Icon from "./Icon";
// import Text from "./Text";

export default {
  title: "Icon",
  component: Icon,
  argTypes: {
    size: {
      control: {
        type: "select",
        options: ["small", "medium", "large"],
      },
    },
    colorIcon: {
      control: {
        type: "background",
      },
    },
  },
} as ComponentMeta<typeof Icon>;

export const All: ComponentStory<typeof Icon> = () => (
  <div className="d-flex flex-wrap">
    <div className="mr-30 mb-20">
      <Icon name="home" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="check" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="bookmark" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="lists" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="more" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="messages" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="notifications" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="profile" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="suscriptions" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="add" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="filter" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="search" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="star" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="send" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="tag" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="dots" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="lock" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="cash" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="arrow-right" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="arrow-left" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="arrow-down" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="arrow-up" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="bell" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="copy" size="large" />
    </div>
    <div className="mr-30">
      <Icon name="chat" size="large" />
    </div>
  </div>
);
