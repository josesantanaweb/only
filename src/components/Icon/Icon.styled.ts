import styled, { css } from "styled-components";
import { IconProps } from "./types";

const IconStyled = styled.i<IconProps>(({ theme, size, colorIcon }) => {
  const gradientMixin = css`
    background: linear-gradient(
      233deg,
      rgba(7, 172, 238, 1) 0%,
      rgba(77, 24, 226, 1) 100%
    );
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  `;
  const colorMixin = css`
    color: ${colorIcon};
  `;
  const sizeMixin = () => {
    switch (size) {
      case "small":
        return css`
          font-size: ${theme.font.sizes.bodySmall};
        `;
      case "medium":
        return css`
          font-size: 24px;
        `;
      case "large":
        return css`
          font-size: ${theme.font.sizes.h3};
        `;
      default:
        return css`
          font-size: ${theme.font.sizes.bodyLarge};
        `;
    }
  };
  return css`
    display: inline-block;
    ${sizeMixin};
    ${colorIcon ? colorMixin : gradientMixin};
  `;
});
export default IconStyled;
