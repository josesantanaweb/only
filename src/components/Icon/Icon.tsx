import React, { FC } from "react";
import IconStyled from "./Icon.styled";
import { IconProps } from "./types";

const Icon: FC<IconProps> = ({ name, size, colorIcon }) => (
  <IconStyled
    className={`icon-${name}`}
    name={name}
    size={size}
    colorIcon={colorIcon}
  />
);

export default Icon;
