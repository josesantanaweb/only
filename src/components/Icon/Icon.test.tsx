import React from "react";
import { render } from "../../test";
import Icon from "./Icon";

describe("Icon", () => {
  it("renders the Icon correctly", () => {
    const { container } = render(<Icon name="home" />);
    expect(container).toMatchSnapshot();
  });

  it("should render with all the props", () => {
    const { container } = render(
      <Icon name="home" size="small" colorIcon="black" />
    );
    expect(container).toMatchSnapshot();
  });
});
