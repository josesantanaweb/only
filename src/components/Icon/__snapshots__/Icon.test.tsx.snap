// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders the Icon correctly 1`] = `
<div>
  <i
    class="sc-bdnxRM fmVSpi icon-home"
  />
</div>
`;

exports[`Icon should render with all the props 1`] = `
<div>
  <i
    class="sc-bdnxRM jAqhZb icon-home"
  />
</div>
`;
